tests/test_schema_syntax.py

```diff
@@ -1346,22 +1419,70 @@ def test_eschema_syntax_link_12(self):
         };
         """
 
-    def test_eschema_syntax_property_13(self):
+    def test_eschema_syntax_link_13(self):
         """
         module test {
-            abstract link union extending except;
+            abstract link union {
+                extending except;
+            };
         };
         """
 
-    def test_eschema_syntax_property_14(self):
+    def test_eschema_syntax_link_14(self):
         """
         module test {
-            abstract link union extending intersect {
+            abstract link union {
+                extending intersect;
                 property except: str;
             };
         };
         """
 
+    def test_eschema_syntax_link_15(self):
+        """
+        module test {
+            abstract link foo {
+                extending bar;
+            };
+        };
+        """
+
+    def test_eschema_syntax_link_16(self):
+        """
+        module test {
+            type foo {
+                required link bar: test::foo {
+                    extending baz;
+                };
+            };
+        };
+
+        """
+
+    def test_eschema_syntax_link_17(self):
+        """
+        module test {
+            type foo {
+                optional link bar: test::foo {
+                    extending baz;
+                };
+            };
+        };
+
+        """
+
+    def test_eschema_syntax_link_18(self):
+        """
+        module test {
+            type foo {
+                link bar: test::foo {
```

Comment:
Also add a test for the case where `link ` is omitted. (I think it will need to be implemented also)